implementation/src/main/java/io/smallrye/mutiny/Uni.java

```diff
@@ -389,6 +393,25 @@ static UniCombine combine() {
      */
     Uni<T> cache();
 
+    // TODO
+    Uni<T> cacheUntil(BooleanSupplier invalidationGuard);
+
```

Comment:
Beside the time-based invalidation, which invalidation cases can you imagine, that are independend of the event itself?
How would the `cacheItemOnly` (resubsceibe on failure) work - ok, maybe that should better be solved with an proper `onFailure` configuration prior the `cache` in the pipe.

Reply:
That's a general-purpose guard mechanism.

Reply:
Note that the current `cache` behaviour is to cache the outcome, no matter if it's an item or a failure. IMHO it's sane.